src/prototypecomponentsnavigatorwindow.h

```diff
@@ -0,0 +1,30 @@
+#ifndef PROTOTYPECOMPONENTSNAVIGATORWINDOW_H
+#define PROTOTYPECOMPONENTSNAVIGATORWINDOW_H
+
+#include <QDialog>
+class QStandardItemModel;
+class QItemSelectionModel;
+
+namespace Ui {
+class PrototypeComponentsNavigatorWindow;
+}
+
+class PrototypeComponentsNavigatorWindow : public QDialog
+{
+    Q_OBJECT
+
+public:
+    explicit PrototypeComponentsNavigatorWindow(QWidget *parent = nullptr);
```

Comment:
Here are no signals and setModel() method, that should be here according to your task. See architectural notes in your ticket in comments.